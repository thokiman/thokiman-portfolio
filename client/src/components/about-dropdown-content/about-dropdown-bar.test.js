import React from "react";
import { shallow } from "enzyme";

import { AboutDropdownBar } from "./about-dropdown-bar.component";

describe("AboutDropdownBar component", () => {
  let wrapper;
  let mockToggleDropdownHidden;
  const mockText = "";
  const mockFalseBool = false;

  beforeEach(() => {
    mockToggleDropdownHidden = jest.fn();

    const mockProps = {
      isDropdownHidden: mockFalseBool,
      toggleDropdownHidden: mockToggleDropdownHidden,
      aboutRoute: mockText,
      timelineCareerRoute: mockText,
      location: { pathname: mockText },
    };
    wrapper = shallow(<AboutDropdownBar {...mockProps} />);
  });

  it("should render AboutDropdownBar component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call toggleDropdownHidden when AboutDropdownLinkContainer 'About' click ", () => {
    wrapper.find("AboutDropdownLinkContainer").at(0).simulate("click");
    expect(mockToggleDropdownHidden).toHaveBeenCalled();
  });
  it("should not render AboutThokimanHeadColorHide if isDropodownHidden is false", () => {
    const mockProps = {
      isDropdownHidden: mockFalseBool,
      toggleDropdownHidden: mockToggleDropdownHidden,
      aboutRoute: mockText,
      timelineCareerRoute: mockText,
      location: { pathname: mockText },
    };
    const newWrapper = shallow(<AboutDropdownBar {...mockProps} />);
    expect(newWrapper.exists("AboutThokimanHeadColorHide")).toBe(false);
  });
});
