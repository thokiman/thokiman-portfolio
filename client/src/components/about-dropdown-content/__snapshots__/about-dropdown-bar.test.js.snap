// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDropdownBar component should render AboutDropdownBar component 1`] = `ShallowWrapper {}`;
